### ISSUE TITLE
(Naomi) The House of the Dead 2, broken 2-Player binds

### ISSUE THREAD

**Issue by Dogway** [libretro/flycast#549]
Input Device: gamepad
Input mapping set to: Auto
Controls set to: Auto
Core Option - Service buttons: Off

When two xbox 360 controllers are plugged, the second player pad Coin button (back button) triggers the service menu. It also inserts coins so there might be some hardcoded input settings on top of RetroArch Auto settings. To expand on that, the 2P Start is also the Test button. And while the 2P Left Stick didn't move the crosshair (after making my way out of the service mode), it could shoot (2P A button) and recharge (2P B button). In service menu 1P Right Stick was being read as the 2P Left Stick.

I don't know if this behavior repeats on other shooters.

**Comment by prince-rakeem** [comment]
I am having a very similar issue using 2 sinden lightguns (port1 and port2 are set to lightguns).  The 2nd player coin insert (which is set to the back right button or num6), inserts a coin AND loads the service menu, therefore resetting the game when you exit service menu.  The only workaround I found was making sure in service menu > coin assignments > coin chute type is set to "common" and not "individual" and disabling coin insert for 2P (setting select to nul in a custom retroarch.cfg for the game).  This doesn't happen with other Naomi games.

**Comment by zaphod4254** [comment]
Yep, same issue.  I have a home arcade cabinet with real coin slots (I keep a bucket of tokens around to use in the coin slots, so it's still "free", just with the more real arcade experience of inserting coins).  I had to set the coin chute type to common and create a game-specific config to remove the coin insert mapping for player 2.  But it sometimes confuses people when they try to insert coins in the 2nd slot for this game and it doesn't do anything.

**Comment by flyinghead** [comment]
This game was the first Naomi game released and it only supports a common coin chute.
From the manual:
![image](https://github.com/libretro/flycast/assets/16261910/bb8176a9-1c56-4b5f-8814-9aa05fe73763)

**Comment by zaphod4254** [comment]
That's not really the point.  I don't mind using common coin chute.  I do for most of my games anyway.  The point is that the COIN 2 input should NOT trigger the Service menu.  It doesn't in DEMUL.

**Comment by zaphod4254** [comment]
There is definitely something cross wired in the code with the Service key and player 2 coin.  Even though the coin 2 doesn't trigger the service menu in any other game, pressing the service button in other games ADDS COINS in those games.  That shouldn't happen either.
